ui/src/helpers/api.js

```diff
@@ -47,7 +50,9 @@ export default class Api {
     validateQuery = async (query) => {
         const form = new FormData();
         form.append('query', query)
-        const response = await this.client.post(`/query/validate`, form);
+        const response = await this.client.post(`/query/validate`, form, {
+            cancelToken: source.token
```

Comment:
why do you need this?
I don't see usage in the source (like source.cancel())

Reply:
@gadotroee I tried a patch like this:
```diff
diff --git a/ui/src/helpers/api.js b/ui/src/helpers/api.js
index abe936a..0caa8ab 100644
--- a/ui/src/helpers/api.js
+++ b/ui/src/helpers/api.js
@@ -48,6 +48,8 @@ export default class Api {
     }
 
     validateQuery = async (query) => {
+        source.cancel();
+
         const form = new FormData();
         form.append('query', query)
         const response = await this.client.post(`/query/validate`, form, {
```

But it prevents HTTP requests altogether. Right now with `cancelToken`, `axios` cancels the previous requests so we only deal with the latest HTTP request.

Reply:
I didn't see any canceled requests when i use it, 
did you see it is really working? and if not maybe this code can be removed

Reply:
@gadotroee OK, I realized my mistake, I'm sorry. Fixed with https://github.com/up9inc/mizu/commit/faaf1c37006d1450cde376cb43fb262d444055e9